server/routerlicious/packages/lambdas-driver/src/document-router/documentLambda.ts

```diff
@@ -55,18 +74,30 @@ export class DocumentLambda implements IPartitionLambda {
 	 * {@inheritDoc IPartitionLambda.handler}
 	 */
 	public handler(message: IQueuedMessage): undefined {
-		if (!this.contextManager.setHead(message)) {
+		this.reprocessingOffset = this.isOffsetWithinReprocessRange(message.offset)
+			? message.offset
+			: undefined;
+		if (!this.contextManager.setHead(message, this.reprocessingOffset)) {
 			this.context.log?.warn(
 				"Unexpected head offset. " +
 					`head offset: ${this.contextManager.getHeadOffset()}, message offset: ${
 						message.offset
 					}`,
 			);
+			// update reprocessRange to avoid reprocessing the same message again
+			if (this.reprocessingOffset) {
```

Comment:
Especially since `reprocessingOffset` is a number, all these `if (this.reprocessingOffset)` checks should specifically be checking `if (this.reprocessingOffset === undefined)` to prevent falsy confusion with `0`